ga4gh/backend.py

```diff
@@ -214,113 +299,44 @@ def variantSetsGenerator(self, request):
             request, self._variantSetIdMap, self._variantSetIds)
 
     def readsGenerator(self, request):
-        # Local utility functions to save some typing
-        def getPosition(readAlignment):
-            return readAlignment.alignment.position.position
-
-        def getEndPosition(readAlignment):
-            return getPosition(readAlignment) + \
-                len(readAlignment.alignedSequence)
-
-        if len(request.readGroupIds) != 1:
-            raise exceptions.NotImplementedException(
-                "Read search over multiple readGroups not supported")
-        readGroupId = request.readGroupIds[0]
-        try:
-            readGroup = self._readGroupIdMap[request.readGroupIds[0]]
-        except KeyError:
-            raise exceptions.ReadGroupNotFoundException(readGroupId)
-        startPosition = request.start
-        equalPositionsToSkip = 0
-        if request.pageToken is not None:
-            startPosition, equalPositionsToSkip = self.parsePageToken(
-                request.pageToken, 2)
+        """
+        Returns a generator over the (read, nextPageToken) pairs defined
+        by the specified request
+        """
+        readGroup = self._getReadGroup(request)
+        startPosition, equalPositionsToSkip = _getIntervalCounters(request)
```

Comment:
Maybe, but we need access to `startPosition` outside `IntervalIterator`, namely to create the iterator.

I think a better design would be to have subclasses of `IntervalIterator`, one for variants and one for reads at present, that could each implement their own methods instead of passing module-level functions into the `IntervalIterator` constructor.  Then `readsGenerator`, etc. would be a one-line function that just called the appropriate `IntervalIterator` subclass. Agree?